samcli/scripts/copy_terraform_built_artifacts.py

```diff
@@ -184,32 +191,60 @@ def search(self, data):
         "--directory",
         type=str,
         required=True,
-        help="directory to which extracted expression " "contents are copied/unzipped to",
+        help="Directory to which extracted expression " "contents are copied/unzipped to",
+    )
+    argparser.add_argument(
+        "--terraform-project-root",
+        type=str,
+        required=False,
+        help="Extracted expression (path) will be relative to the terraform project root if provided.",
     )
 
     arguments = argparser.parse_args()
     directory_path = os.path.abspath(arguments.directory)
+    terraform_project_root = os.path.abspath(arguments.terraform_project_root)
+
+    extracted_attribute_path = None
+    data_object = None
 
     if not os.path.exists(directory_path) or not os.path.isdir(directory_path):
-        raise OSError("Expected --directory-path to be a valid directory")
+        LOG.error("Expected --directory to be a valid directory!")
+        cli_exit()
+    if terraform_project_root:
+        if not os.path.exists(terraform_project_root) or not os.path.isdir(terraform_project_root):
+            LOG.error("Expected --terraform-project-path to be a valid directory!")
+            cli_exit()
 
     # Load and Parse
-    data_object = json.load(sys.stdin)
-    output_path = Parser(expression=arguments.expression).parse().search(data=data_object)
+    try:
+        data_object = json.load(sys.stdin)
+    except ValueError as ex:
+        LOG.error("Parsing JSON from stdin unsuccessful!", exc_info=True)
+        cli_exit()
 
+    try:
+        extracted_attribute_path = Parser(expression=arguments.expression).parse().search(data=data_object)
+    except ResolverException as ex:
+        LOG.error(ex.message, exc_info=True)
+        cli_exit()
     # Check if that is indeed a path.
-    if not os.path.exists(output_path):
-        raise OSError("Extracted attribute path from provided expression does not exist!")
-    filepath = os.path.abspath(output_path)
-
-    # Unzip the zipped file or copy the dir
+    abs_attribute_path = (
+        os.path.abspath(os.path.join(terraform_project_root, extracted_attribute_path))
+        if terraform_project_root
```

Comment:
do we need to check if the `extracted_attribute_path` is relative path before we join it with the `terraform_project_root` ?

Reply:
Sure!

```python
        os.path.abspath(os.path.join(terraform_project_root, extracted_attribute_path))
        if terraform_project_root and not os.path.isabs(extracted_attribute_path)
        else os.path.abspath(extracted_attribute_path)
```

Something like that should be good.